### PR TITLE
Enable HTML test report

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Enable HTML test report for easier debug in case of test failure
  - Test output will be available in the page for failed test cases.
- The result would be uploaded as artifact and can be downloaded as ZIP file
- The report will be stored in separate folder per package

Sample Image:
Screenshot 2024-11-12 at 16 38 26

### Related Issues
n/a

### Check List
- [ ] Updated documentation (docs/ppl-lang/README.md)
- [ ] Implemented unit tests
- [ ] Implemented tests for combination with other commands
- [ ] New added source code should include a copyright header
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
@ykmr1224 very nice !!

File: build.sbt [hunk 1]
```diff
@@ -53,7 +53,11 @@ lazy val commonSettings = Seq(
   compileScalastyle := (Compile / scalastyle).toTask("").value,
   Compile / compile := ((Compile / compile) dependsOn compileScalastyle).value,
   testScalastyle := (Test / scalastyle).toTask("").value,
+  // Enable HTML report and output to separate folder per package
+  Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-h", s"target/test-reports/${name.value}"),
   Test / test := ((Test / test) dependsOn testScalastyle).value,
+  // Needed for HTML report
+  libraryDependencies += "com.vladsch.flexmark" % "flexmark-all" % "0.64.8",
```
Comment (dai-chen) [inline build.sbt:60]: should we make this test scope?

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by Swiddis — APPROVED** [review]

**Review by seankao-az — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (ykmr1224) [reply build.sbt]: Good catch. Fixed this.

**Pull request merged**